src/GeniusYield/Test/Utils.hs

```diff
@@ -387,17 +236,20 @@ pattern (:=) x y = (x, y)
 
 infix 0 :=
 
--------------------------------------------------------------------------------
--- Preset StdGen
--------------------------------------------------------------------------------
-
-pureGen :: StdGen
-pureGen = mkStdGen 42
-
-{- -----------------------------------------------------------------------------
-  CLB
------------------------------------------------------------------------------ -}
+{- | Utilizing 'ExceptT' as a "eager monad" transformer.
 
--- | Variant of `logInfo` from @Clb@ that logs a string with @Info@ severity.
-logInfoS :: Monad m => String -> Clb.ClbT m ()
-logInfoS s = Clb.logInfo $ Clb.LogEntry Clb.Info s
+'Left' does not indicate failure, rather it indicates that "target value has been obtained"
+and that we can exit eagerly.
+-}
+type EagerT m a = ExceptT a m ()
```

Comment:
Surely it works, but it is counter-intuitive. I notice that it's (`runEagerT`) only being used here at two places, where I think this short-circuiting behaviour is not required?

Reply:
Yeah I wish we had a specific "early exit" monad with its own set of utilities in base. Don't like using the term `Except` even though it's functionally the same.